src/main/java/com/booleanuk/Poker.java

```diff
@@ -1,27 +1,178 @@
+                if (currentThreeOfAKind == null ||
+                        Integer.parseInt(currentThreeOfAKind[0]) < Integer.parseInt(handsWithValues[i][0])) {
+                    currentThreeOfAKind = new String[]{handsWithValues[i][0], handsWithValues[i][1], handsWithValues[i][2]};
+                    winningHandIndex = i;
+                }
+            }
+            // Check for pair in hands that have three cards
+            else if (handsWithValues[i].length == 3) {
+                if(Objects.equals(handsWithValues[i][0], handsWithValues[i][1])){
+                    if (currentHighestPair == null ||
+                            Integer.parseInt(currentHighestPair[0]) < Integer.parseInt(handsWithValues[i][0])) {
+                        currentHighestPair = new String[]{handsWithValues[i][0], handsWithValues[i][1]};
+                        winningHandIndex = i;
+                    }
+                }
+                if(Objects.equals(handsWithValues[i][0], handsWithValues[i][2])){
+                    if (currentHighestPair == null ||
+                            Integer.parseInt(currentHighestPair[0]) < Integer.parseInt(handsWithValues[i][0])) {
+                        currentHighestPair = new String[]{handsWithValues[i][0], handsWithValues[i][1]};
+                        winningHandIndex = i;
+                    }
+                }
+                if(Objects.equals(handsWithValues[i][1], handsWithValues[i][2])){
+                    if (currentHighestPair == null ||
+                            Integer.parseInt(currentHighestPair[0]) < Integer.parseInt(handsWithValues[i][1])) {
+                        System.out.println("NEW HIGHEST PAIR IN HAND SIZE OF 3");
+                        currentHighestPair = new String[]{handsWithValues[i][0], handsWithValues[i][1]};
+                        winningHandIndex = i;
+                    }
+                }
+            }
+            // Check for pairs
+            if (handsWithValues[i].length == 2 && Objects.equals(handsWithValues[i][0], handsWithValues[i][1])) {
+                System.out.println("HAND " + i + " HAS A PAIR");
+                if (currentHighestPair == null ||
+                        Integer.parseInt(currentHighestPair[0]) < Integer.parseInt(handsWithValues[i][0])) {
+                    currentHighestPair = new String[]{handsWithValues[i][0], handsWithValues[i][1]};
+                    winningHandIndex = i;
+                }
+            }
+        }
+
+
+        // If no three of a kind or pairs, return empty
+        if (winningHandIndex == -1) {
+            return new String[]{};
+
+        }
+
+        if(currentHighestPair == null && currentThreeOfAKind == null){
+            return new String[]{};
+        }
+
+        return hands[winningHandIndex];
     }
 
+
+
     // Extension 3
     public String extensionThreeMethods() {
         // Completely replace this method with suitable methods to solve Extension 3
```

Comment:
Your approach using a hashmap for card values and handling different hand scenarios is well thought out. Also, good work distinguishing between pairs and three-of-a-kind.